### PR TITLE
[WIP] split csr2csc function

### PR DESCRIPTION
**Pull request by ashokei**

### CODE REVIEW

File: samples/deeplearning/sparse_adagrad_fused/sparse_adagrad.cpp [hunk 1]
```diff
@@ -513,6 +517,21 @@ int main(int argc, char * argv[]) {
       tU += eio[j][i]->U;
     }
   }
+  double csr2csc_time = get_time();
+  CTraceEvents::Scoped cevent("csr2csc", "", "", "", true);
```
Comment (alheinecke) [inline samples/deeplearning/sparse_adagrad_fused/sparse_adagrad.cpp:521]: why not using libxsmm timer which is available?

File: samples/deeplearning/sparse_adagrad_fused/sparse_adagrad.cpp [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #endif
 
 #include "radix_sort.h"
+#include "ctevents.h"
```
Comment (alheinecke) [inline samples/deeplearning/sparse_adagrad_fused/sparse_adagrad.cpp:40]: This include and our libraries needs to use are missing in this PR.

**Review by alheinecke — CHANGES_REQUESTED** [review]
please see detailed comments

File: samples/deeplearning/sparse_adagrad_fused/sparse_adagrad.cpp [hunk 3]
```diff
@@ -513,6 +517,21 @@ int main(int argc, char * argv[]) {
...
+      csr2csc(eio[j][i]);
+    }
+  }
+  csr2csc_time = get_time()-csr2csc_time;
+  printf("csr2csc time = %.3f ms per table, %.3f total\n", csr2csc_time/(LS*iters*K), csr2csc_time/(iters*K));
+
+return 1;
   int warmup = (iters > 2 ? 2 : iters);
 
   for(int i = 0; i < warmup; i++) {
```
Comment (ashokei) [inline samples/deeplearning/sparse_adagrad_fused/sparse_adagrad.cpp:537]: saiusJK

**Comment by alheinecke** [comment]
Is this still active and should be merged it has now additional conflicts.

**Pull request closed without merging**